homeassistant/components/modbus/validators.py

```diff
@@ -271,10 +271,6 @@ def duplicate_entity_validator(config: dict) -> dict:
                     a = str(entry[CONF_TARGET_TEMP])
                     a += "_" + str(inx)
                     entry_addrs.add(a)
-                if CONF_HVAC_MODE_REGISTER in entry:
```

Comment:
One register in one device can only belong to one entity. Allowing one register to belong to multiple entities, would cause a lot of update problems !

Apart from that please explain why one register need to be allowed from multiple entities, it is not logical !

Your fix, only removes the check, but you miss the more important part to update all entities using this register. 

Currently we read one entity at the time, and update it....but you want to turn that around to when reading a register update n entities.

Reply:
I already described that in the linked issue. The situation is that with 2023.12 and before this was possible. I had a floor heating which only have one mode register for all rooms. So in the past I changed it on one climate card and the others cards got updated with the next update from modbus. This was fine for me. Every card allows changing the overall mode.

With 2024.1.0 the mentioned code changes were added which makes this not possible anymore.

Reply:
That might be, but you still need to address the issue of updating multiple entities when receiving a register value, and that is a major change of the update methods.

Apart from that I personally still feel it is wrong to use one register in multiple entities, but this is open source so my only concern is to secure your patch works correctly.